tests/NuGet.Services.AzureSearch.Tests/DownloadTransferrerFacts.cs

```diff
@@ -28,6 +28,197 @@ public void ReturnsEmptyResult()
+            public void SupportsZeroPopularityTransfer()
+            {
+                PopularityTransfer = 0;
+
+                DownloadData.SetDownloadCount("A", "1.0.0", 100);
+                DownloadData.SetDownloadCount("B", "1.0.0", 5);
+
+                AddPopularityTransfer("A", "B");
+
+                var result = Target.InitializeDownloadTransfers(
+                    DownloadData,
+                    PopularityTransfers,
+                    DownloadOverrides);
+
+                Assert.Equal(new[] { "A", "B" }, result.Keys);
+                Assert.Equal(100, result["A"]);
+                Assert.Equal(5, result["B"]);
+            }
+
+            [Fact]
+            public void PackageWithOutgoingTransferRejectsIncomingTransfer()
+            {
+                PopularityTransfer = 1;
+
+                DownloadData.SetDownloadCount("A", "1.0.0", 100);
+                DownloadData.SetDownloadCount("B", "1.0.0", 0);
+                DownloadData.SetDownloadCount("C", "1.0.0", 0);
+
+                AddPopularityTransfer("A", "B");
+                AddPopularityTransfer("A", "C");
+                AddPopularityTransfer("B", "C");
+
+                var result = Target.InitializeDownloadTransfers(
+                    DownloadData,
+                    PopularityTransfers,
+                    DownloadOverrides);
+
+                // B has incoming and outgoing popularity transfers. It should reject the incoming transfer.
+                Assert.Equal(new[] { "A", "B", "C" }, result.Keys);
+                Assert.Equal(0, result["A"]);
+                Assert.Equal(0, result["B"]);
+                Assert.Equal(50, result["C"]);
+            }
+
+            [Fact]
+            public void PopularityTransfersAreNotTransitive()
+            {
+                PopularityTransfer = 1;
+
+                DownloadData.SetDownloadCount("A", "1.0.0", 100);
+                DownloadData.SetDownloadCount("B", "1.0.0", 100);
+                DownloadData.SetDownloadCount("C", "1.0.0", 100);
+
+                AddPopularityTransfer("A", "B");
+                AddPopularityTransfer("B", "C");
+
+                var result = Target.InitializeDownloadTransfers(
+                    DownloadData,
+                    PopularityTransfers,
+                    DownloadOverrides);
+
+                // A transfers downloads to B.
+                // B transfers downloads to C.
+                // B and C should reject downloads from A.
+                Assert.Equal(new[] { "A", "B", "C" }, result.Keys);
+                Assert.Equal(0, result["A"]);
+                Assert.Equal(0, result["B"]);
+                Assert.Equal(200, result["C"]);
+            }
+
+            [Fact]
+            public void RejectsCyclicalPopularityTransfers()
+            {
+                PopularityTransfer = 1;
+
+                DownloadData.SetDownloadCount("A", "1.0.0", 100);
+                DownloadData.SetDownloadCount("B", "1.0.0", 100);
```

Comment:
It's possible to "lose" a download, right? Like if the percentage against the total does no result in an exact number. We should have tests around that. It's okay to lose a download here or there but the behavior should be asserted IMO

Reply:
Yup, that's possible. I'll add test coverage 👍

Reply:
Added